src/app_service/service/network_connection/service.nim

```diff
@@ -0,0 +1,269 @@
+import NimQml, chronicles, Tables, strutils, sequtils, sugar, strformat, json
+
+import ../../../app/global/global_singleton
+import ../../../app/core/eventemitter
+import ../../../app/core/signals/types
+import ../../../app/core/tasks/[qt, threadpool]
+
+import ../collectible/service as collectible_service
+import ../wallet_account/service as wallet_service
+import ../network/service as network_service
+import ../node/service as node_service
+import ../../../backend/backend as backend
+
+logScope:
+  topics = "network-connection-service"
+
+type ConnectionState* {.pure.} = enum
+  Successful = 0,
+  Failed = 1,
+  Retrying = 2
+
+type ConnectionStatus* = ref object of RootObj
+    connectionState*: ConnectionState
+    completelyDown*: bool
+    chainIds*: seq[int]
+    lastCheckedAt*: int
+    timeToAutoRetryInSecs*: int
+    timer*: QTimer
+    withCache*: bool
+
+const SIGNAL_CONNECTION_UPDATE* = "signalConnectionUpdate"
+const SIGNAL_REFRESH_COLLECTIBLES* = "signalRefreshCollectibles"
+
+type NetworkConnectionsArgs* = ref object of Args
+  website*: string
+  completelyDown*: bool
+  connectionState*: ConnectionState
+  chainIds*: string
+  lastCheckedAt*: int
+  timeToAutoRetryInSecs*: int
+  withCache*: bool
+
+type RetryCollectibleArgs* = ref object of Args
+   addresses*: seq[string]
+
+const BLOCKCHAINS* = "blockchains"
+const MARKET* = "market"
+const COLLECTIBLES* = "collectibles"
+const BACKOFF_TIMERS* = [30, 60, 180, 600, 3600, 10800]
+
+include  ../../common/json_utils
+
+proc newConnectionStatus(): ConnectionStatus =
+  return ConnectionStatus(
+    connectionState: ConnectionState.Successful,
+    completelyDown: false,
+    chainIds: @[],
+    lastCheckedAt: 0,
+    timeToAutoRetryInSecs: BACKOFF_TIMERS[0],
+    timer: newQTimer(),
+    withCache: false
+  )
+
+QtObject:
+  type Service* = ref object of QObject
+    closingApp: bool
+    events: EventEmitter
+    threadpool: ThreadPool
+    walletService: wallet_service.Service
+    networkService: network_service.Service
+    collectibleService: collectible_service.Service
+    nodeService: node_service.Service
+    connectionStatus: Table[string, ConnectionStatus]
+
+  # Forward declaration
+  proc checkConnected(self: Service)
+
+  proc delete*(self: Service) =
+    self.closingApp = true
+    self.QObject.delete
+
+  proc newService*(
+    events: EventEmitter,
+    threadpool: ThreadPool,
```

Comment:
@Khushboo-dev-cpp I don't see that you're using threadpool here (if so, feel free to remove it)?